### PR TITLE
Parse SocketAddrV6::scope_id

### PR DESCRIPTION
**Pull request by tamird**
r? @dtolnay

### CODE REVIEW

File: library/std/src/net/parser.rs [hunk 1]
```diff
@@ -90,7 +92,7 @@ impl<'a> Parser<'a> {
     {
         self.read_atomically(move |p| {
             if index > 0 {
-                let _ = p.read_given_char(sep)?;
+                let () = p.read_given_char(sep)?;
```
Comment (dtolnay) [inline library/std/src/net/parser.rs:95]:
```suggestion
                p.read_given_char(sep)?;
```

**Review by dtolnay — COMMENTED** [review]
Looks good.

All the `let () =` seem unnecessarily to me though.

File: library/std/src/net/parser.rs (hunk 1 above)
Comment (tamird) [reply library/std/src/net/parser.rs:95]: These are intentional; without them, a missing `?` at the end of the line would go unnoticed. See https://github.com/rust-lang/rust-clippy/issues/3915

Comment (dtolnay) [reply library/std/src/net/parser.rs:95]: Okay but this conflicts with https://rust-lang.github.io/rust-clippy/rust-1.46.0/index.html#let_unit_value. :slightly_frowning_face: I am not a fan. Could we protect against missing `?` by using #\[must_use\] on the relevant functions instead? Someone adding new calls isn't going to remember to write `let ()` if they don't remember to write `?`.

Comment (tamird) [reply library/std/src/net/parser.rs:95]: You're the boss. Done.

**Review by dtolnay — APPROVED** [review]
Thanks!

**Pull request merged**